contrib/terraform/gcp/README.md

```diff
@@ -0,0 +1,163 @@
+* Login to the kubespray ansible instance and execute the **cluster.yml** ansible-playbook to setup cluster. The cluster execution details will be shown in terminal and also the output is redirected to the **“output”** file(10).
+* Kubespray-Ansible Machine Ip and One of the Kube-master Ip will be displayed as part of output. 
+
+
+## Status
+This will install a Kubernetes cluster on GCP. 
+
+## Approach
+The terraform configuration inspects variables found in [variables.tf](variables.tf) to create resources in GCP.
+
+The terraform script will take care of creating Master nodes,etcd nodes,worker nodes/minions, kubespray-ansible-node based on the configuration details. 
+
+There is a [python script](./kube_configurations/GenerateInventoryFile.py) that generates a dynamic inventory that is consumed by the kubespray cluster.yml.
+
+### Kubernetes Nodes
+You can create different kubernetes topologies by setting the below mentioned variable to indicate number of hosts.
+
+- Master nodes: `kube_master_target_size` variable
+- Etcd nodes : `kube_etcd_target_size` variable
+- Kubernetes worker nodes or minions: `kube_minion_target_size` variable
+
+- Kubespray ansible node: `kube_ansible_target_size` variable
+
+Note that the kubespray Ansible script will report an invalid configuration if you wind up
+with an *even number* of etcd instances since that is not a valid configuration. Also it is recommended to have multiple master nodes for high availability.
+
+## Prerequisites
+
+- [Install Terraform](https://www.terraform.io/intro/getting-started/install.html)
+- [Service account key json](https://cloud.google.com/iam/docs/creating-managing-service-account-keys) (Appropirate roles to be assinged to service account to create computes)
+- Centos or Redhat os to be used for kubespray ansible instance.
+- You have a pair of keys generated which is the part of image that can be used to secure the new hosts.
+- [Kubespray](https://github.com/kubernetes-sigs/kubespray.git). Make sure the project name is kubespray.
+
+
+### Configuration
+
+#### Service Account key json file
+Details from service account key json file should be updated in the [account.json](./clustertfiles/gcp-auth-config-json/account.json) file
+The Project ID associated with the service account key should be set in the [variables.tf](variables.tf)
+
+Note: To deploy several clusters within the same project you need to use [terraform workspace](https://www.terraform.io/docs/state/workspaces.html#using-workspaces).
+
+#### SSH Key Setup
+SSH keypair is required by Kubespray-Ansible to access the newly provisioned Instances on GCP.When ssh keys are generated for the user 
+
+* private key to be placed in the file [key](./kube_configurations/ssh-key).
+* Public key to be added at the project wide level. [Refer link to add-ssh-key at project level](https://cloud.google.com/compute/docs/instances/adding-removing-ssh-keys) 
+
+#### Cluster variables:
```

Comment:
How does one change things in the Ansible variables? Like using containerd instead of docker, does that require to have a fork?

Reply:
ssh-key is the private key of the user defined in the Terraform tfvars file. Basically this user is used to setup cluster.